File: app/src/main/java/com/example/schooldiary/ui/adapters/diary/SubjectHolder.java
package com.example.schooldiary.ui.adapters.diary;

import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.schooldiary.R;
import com.example.schooldiary.databinding.ItemSubjectBinding;
import com.example.schooldiary.model.Subject;

public class SubjectHolder extends RecyclerView.ViewHolder {

    private View itemView;
    private SubjectClickListener listener;

    public interface SubjectClickListener {
        void openFragment(int id, String name, String homework, String classroom, String teacher);
    }

    public SubjectHolder(@NonNull View itemView, SubjectClickListener listener) {
        super(itemView);
        this.itemView = itemView;
        this.listener = listener;
    }

    public void bindData(Subject subject) {

        ItemSubjectBinding binding = ItemSubjectBinding.bind(itemView);
        TextView startTime = binding.subjectStartTime;
        TextView subjectName = binding.subjectName;
        TextView subjectAuditory = binding.subjectAuditory;

        startTime.setText(calculateTime(subject.getSerialNumber()));
        subjectName.setText(subject.getName());
        subjectAuditory.setText("238");

        itemView.setOnClickListener(view -> listener.openFragment(R.id.show_subject_fragment,
                subject.getName(), subject.getHomework(), "238", "Gorina Nonna Garikovna"));
    }

    private String calculateTime(int number) {
        switch (number) {
            case 1:
                return "08:00";
            case 2:
                return "08:55";
            case 3:
                return "09:50";
            case 4:
                return "10:55";
            case 5:
                return "12:00";
            case 6:
                return "13:00";
            case 7:
                return "13:55";
            case 8:
                return "14:50";
            default:
                return "00:00";
        }
    }
}

File: app/src/main/java/com/example/schooldiary/ui/adapters/diary/SubjectHolder.kt
package com.example.schooldiary.ui.adapters.diary

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.example.schooldiary.R
import com.example.schooldiary.databinding.ItemSubjectBinding
import com.example.schooldiary.model.Subject

class SubjectHolder(itemView: View, private val listener: SubjectClickListener) : RecyclerView.ViewHolder(itemView) {

    interface SubjectClickListener {
        fun openFragment(id: Int, name: String, homework: String, classroom: String, teacher: String)
    }

    fun bindData(subject: Subject) {
        val binding = ItemSubjectBinding.bind(itemView)
        binding.startTime.text = calculateTime(subject.serialNumber ?: 0)
        binding.name.text = subject.name ?: UNKNOWN
        // Firebase still doesn't have the classroom number and teacher's name
        binding.classroom.text = "557"
        itemView.setOnClickListener {
            listener.openFragment(R.id.show_subject_fragment,
                    subject.name ?: UNKNOWN,
                    subject.homework ?: UNKNOWN,
                    "557",
                    "Birukova Tatyana Leontyevna")
        }
    }

    private fun calculateTime(number: Int): String {
        return when (number) {
            1 -> "08:00"
            2 -> "08:55"
            3 -> "09:50"
            4 -> "10:55"
            5 -> "12:00"
            6 -> "13:00"
            7 -> "13:55"
            8 -> "14:50"
            else -> "00:00"
        }
    }

    companion object {
        private const val UNKNOWN = "Unknown"
    }
}